src/js/components/LegendPanel/LegendPanel.js

```diff
@@ -67,36 +51,114 @@ export default class LegendPanel extends Component {
       layerIds = layerIds.concat(graphicsLayerIds);
     }
 
-    //- Get layers from the webmap, we could comment out this block but may miss any layers added from
-    //- the webmap as graphics or feature layers since those won't be in layerIds
-    // legendInfos = webmapInfo.operationalLayers.filter((item) => {
-    //   //- Add them to ignores so they do not show up twice
-    //   if (item.layerObject) { ignores.push(item.id); }
-    //   return item.layerObject;
-    // }).map((layer) => {
-    //   return {
-    //     layer: layer.layerObject,
-    //     title: '' // layer.layerObject.name
-    //   };
-    // });
-
     if (layerIds) {
       //- Remove layers to ignore
       ids = layerIds.filter(id => ignores.indexOf(id) === -1);
       ids.forEach((layerId) => {
         const layer = map.getLayer(layerId);
         if (layer) {
-          legendInfos.push({ layer, title: '' });
+          legendInfos.push({ layer });
         }
       });
     }
 
+    function compare(a, b) {
+      if(a.layer.order === undefined) {
+        a.layer.order = 0;
+      }
+      if(b.layer.order === undefined) {
+        b.layer.order = 0;
+      }
+      if(a.layer.order < b.layer.order) {
+        return 1;
+      }
+      if(a.layer.order > b.layer.order) {
+        return -1;
+      }
+      return 0;
+    }
+    legendInfos.sort(compare);
+
     return legendInfos;
   }
 
+  createLegend = (layerDiv, index) => {
+    let childComponent;
+    const {map, language, settings} = this.context;
```

Comment:
So, if we are grabbing all of these heavy things from context just to pass them down to this same single childComponent, we would be better served grabbing them from context one step lower, in that childComponent, and avoiding all of these unnecessary props.